diff -up irqbalance-1.0.4/irqbalance.1.orig irqbalance-1.0.4/irqbalance.1
--- irqbalance-1.0.4/irqbalance.1.orig	2012-08-27 17:26:39.000000000 +0200
+++ irqbalance-1.0.4/irqbalance.1	2014-03-26 16:34:47.229155386 +0100
@@ -57,6 +57,8 @@ irq is balanced, but the assigned object
 .P
 .I ignore
 irq affinity hint value is completely ignored
+.P
+The default value for hintpolicy is ignore
 
 .TP
 .B --powerthresh=<threshold>
diff -up irqbalance-1.0.4/irqbalance.c.orig irqbalance-1.0.4/irqbalance.c
--- irqbalance-1.0.4/irqbalance.c.orig	2012-08-27 17:26:39.000000000 +0200
+++ irqbalance-1.0.4/irqbalance.c	2014-03-26 16:36:57.361401164 +0100
@@ -48,7 +48,7 @@ int foreground_mode;
 int numa_avail;
 int need_rescan;
 extern cpumask_t banned_cpus;
-enum hp_e hint_policy = HINT_POLICY_SUBSET;
+enum hp_e hint_policy = HINT_POLICY_IGNORE;
 unsigned long power_thresh = ULONG_MAX;
 unsigned long long cycle_count = 0;
 char *pidfile = NULL;
