Backport of following upstream commit:

c75c1511c87defc22a083460802e9adc5af49cf0
--
diff -up irqbalance-1.0.4/classify.c.orig irqbalance-1.0.4/classify.c
--- irqbalance-1.0.4/classify.c.orig	2014-05-29 12:06:58.672239564 +0200
+++ irqbalance-1.0.4/classify.c	2014-05-29 12:09:16.032550034 +0200
@@ -375,9 +375,13 @@ struct irq_info *add_new_irq(int irq)
 	struct irq_info *new, *nnew;
 
 	new = calloc(sizeof(struct irq_info), 1);
+	if (!new)
+		return NULL;
 	nnew = calloc(sizeof(struct irq_info), 1);
-	if (!new || !nnew)
+	if (!nnew) {
+		free(new);
 		return NULL;
+	}
 
 	new->irq = irq;
 	new->type = IRQ_TYPE_LEGACY;
@@ -418,6 +422,10 @@ void migrate_irq(GList **from, GList **t
 
 	find.irq = info->irq;
 	entry = g_list_find_custom(*from, &find, compare_ints);
+
+	if (!entry)
+		return;
+
 	tmp = entry->data;
 	*from = g_list_delete_link(*from, entry);
 
diff -up irqbalance-1.0.4/numa.c.orig irqbalance-1.0.4/numa.c
--- irqbalance-1.0.4/numa.c.orig	2012-08-27 17:26:39.000000000 +0200
+++ irqbalance-1.0.4/numa.c	2014-05-29 12:09:16.032550034 +0200
@@ -64,6 +64,10 @@ static void add_one_node(const char *nod
 		return;
 	sprintf(path, "%s/%s/cpumap", SYSFS_NODE_PATH, nodename);
 	f = fopen(path, "r");
+	if (!f) {
+		free(new);
+		return;
+	}
 	if (ferror(f)) {
 		cpus_clear(new->mask);
 	} else {
