diff -up irqbalance-1.0.7/Makefile.am.orig irqbalance-1.0.7/Makefile.am
--- irqbalance-1.0.7/Makefile.am.orig	2015-04-01 12:19:47.692136440 +0200
+++ irqbalance-1.0.7/Makefile.am	2015-04-01 12:21:14.577286143 +0200
@@ -25,7 +25,7 @@ ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST = COPYING autogen.sh misc/irqbalance.service misc/irqbalance.env
 
 INCLUDES = -I${top_srcdir} 
-AM_CFLAGS = -g -Os -W -Wall -Wshadow -Wformat -Wundef -Wl,-z,relro,-z,now -pie -fpie $(GLIB_CFLAGS) -D_GNU_SOURCE $(LIBCAP_NG_CFLAGS)
+AM_CFLAGS = -g -Os -W -Wall -Wshadow -Wformat -Wundef -Wl,-z,relro,-z,now -pie -fpie $(GLIB_CFLAGS) -D_GNU_SOURCE $(LIBCAP_NG_CFLAGS) -DINCLUDE_BANSCRIPT
 AM_CPPFLAGS = -W -Wall -Wshadow -Wformat -Wundef -D_GNU_SOURCE
 noinst_HEADERS = bitmap.h constants.h cpumask.h irqbalance.h non-atomic.h \
 	types.h
